src/main/java/com/jagrosh/jmusicbot/utils/TimeUtil.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 John Grosh <john.a.grosh@gmail.com>.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jagrosh.jmusicbot.utils;
+
+public class TimeUtil
+{
+
+    public static String formatTime(long duration)
+    {
+        if(duration == Long.MAX_VALUE)
+            return "LIVE";
+        long seconds = Math.round(duration/1000.0);
+        long hours = seconds/(60*60);
+        seconds %= 60*60;
+        long minutes = seconds/60;
+        seconds %= 60;
+        return (hours>0 ? hours+":" : "") + (minutes<10 ? "0"+minutes : minutes) + ":" + (seconds<10 ? "0"+seconds : seconds);
+    }
+
+    /**
+     * @param args timestamp formatted as: [+ | -] &lt;HH:MM:SS | MM:SS | SS&gt;
+     * @return Time in milliseconds, negative if seeking backwards relatively
+     */
+    public static SeekTime parseTime(String args)
+    {
+        if (args.length() == 0 || args.length() > 8) return null;
```

Comment:
I also believe that any amount of beginning zeros should be accepted. Beginning zeros should not interfere seeking. In fact, seeking half a second back would also be nice.

Reply:
> Although unlikely, a song or track could last hundreds of hours, in which case seeking to `105:12:15` would be impossible. I don't think the args length limit is needed here, as the integer parsing later will handle invalid inputs.

I agree.

> I also believe that any amount of beginning zeros should be accepted. Beginning zeros should not interfere seeking. In fact, seeking like 3 and half a seconds would also be nice as an example.

Seeking with decimal numbers would be nice, I agree. Coming to think of it, it might be even nicer if you could use any float number as argument. As long as it adheres to the x:x:x format, I think it would be fine.